cns/nodesubnet/ip_fetcher.go

```diff
@@ -0,0 +1,77 @@
+package nodesubnet
+
+import (
+	"context"
+	"log"
+	"net"
+	"time"
+
+	"github.com/Azure/azure-container-networking/nmagent"
+	"github.com/pkg/errors"
+)
+
+var ErrorRefreshSkipped = errors.New("refresh skipped due to throttling")
+
+// This interface is implemented by the NMAgent Client, and also a mock client for testing
+type InterfaceRetriever interface {
+	GetInterfaceIPInfo(ctx context.Context) (nmagent.Interfaces, error)
+}
+
+type IPFetcher struct {
+	// Node subnet state
+	secondaryIPQueryInterval   time.Duration // Minimum time between secondary IP fetches
+	secondaryIPLastRefreshTime time.Time     // Time of last secondary IP fetch
+
+	ipFectcherClient InterfaceRetriever
+}
+
+func NewIPFetcher(nmaClient InterfaceRetriever, queryInterval time.Duration) *IPFetcher {
+	return &IPFetcher{
+		ipFectcherClient:         nmaClient,
+		secondaryIPQueryInterval: queryInterval,
+	}
+}
+
+// If secondaryIPQueryInterval has elapsed since the last fetch, fetch secondary IPs
```

Comment:
This should also go in the body of the method so as not to accidentally appear in godoc